### PR TITLE
Fix parsing dict, upgrade select, fix test error

### LINKED ISSUE #5361: Upgrade Select in Diagram

### PR DESCRIPTION
**Pull request by matborowczyk**
# Description

Sanitise is added only on shapes that go through Form, as that the point where dict are getting "stringified".

closes #5361 #5356 #5354 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Code is clear and sufficiently documented
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

File: src/UI/Components/Diagram/components/FormModal.tsx [hunk 1]
```diff
@@ -158,7 +170,7 @@ const FormModal = ({
     const tempPossibleForms = getOptions(services, [
       {
         key: "default_option",
-        value: "Choose a Service",
+        value: undefined,
```
Comment (LukasStordeur) [inline src/UI/Components/Diagram/components/FormModal.tsx:173]: Is this to be able to clear the select? If that's the case, default_option doesn't seem like the right key to me. Also, I wonder if we really do need that option at all

File: src/UI/Components/Diagram/components/FormModal.tsx [hunk 2]
```diff
@@ -225,21 +235,19 @@ const FormModal = ({
...
-            isDisabled={cellView !== null}
           >
             {possibleForms.map(({ key, value }) => (
-              <SelectOption
-                key={key}
-                value={value}
-                isPlaceholder={value === "Choose a Service"}
-              />
+              <SelectOption key={key} value={value}>
+                {value || "Choose a Service"}
```
Comment (LukasStordeur) [inline src/UI/Components/Diagram/components/FormModal.tsx:249]: I think having a cross icon in the select toggle to clear the value is better than having an option displaying as Choose a Service

File: src/UI/Components/Diagram/components/FormModal.tsx (hunk 1 above)
Comment (matborowczyk) [reply src/UI/Components/Diagram/components/FormModal.tsx]: After some thought I agree, removed

File: src/UI/Components/Diagram/components/FormModal.tsx [hunk 3]
```diff
@@ -68,14 +71,24 @@ const FormModal = ({
...
+
   const onEntityChosen = useCallback(
-    (
-      _event,
-      value: string | SelectOptionObject,
-      isPlaceholder: boolean | undefined,
-      possibleForms: PossibleForm[],
-    ) => {
-      if (isPlaceholder) {
+    (value: string | number | undefined, possibleForms: PossibleForm[]) => {
```
[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/UI/Components/Diagram/components/FormModal.tsx]: That was due to the typing that are from the onSelect in the components itself, but changed that now to casing in the component

**Pull request closed without merging**